x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/actions/email/ExecutableEmailAction.java

```diff
@@ -25,9 +25,12 @@
 import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
 
 public class ExecutableEmailAction extends ExecutableAction<EmailAction> {
 
+    private static final AtomicInteger counter = new AtomicInteger(0);
```

Comment:
I know integer max is pretty large w/r/t sending emails ... but wonder if appending a random UUID would be better suited to ensure uniqueness to avoid exceeding the max ?

Would org.elasticsearch.common.RandomBasedUUIDGenerator#getBase64UUID() be a candidate instead ?

(I tested it looks like if you exceed max it just rolls over to negative numbers, but that is likely an implementation detail since it is not the doc)

Reply:
I considered a random UUID overkill (slow to generate), as creation seemed more expensive to me, and the message id already contained the action id and the `Wid` (which is unique per watch execution), so no need for more data from my point of view.

I also was not worried about going negative, just another change in the message id.

Rereading [RFC 2822](https://tools.ietf.org/html/rfc2822.html) however states that the message id looks like this

```
msg-id          =       [CFWS] "<" id-left "@" id-right ">" [CFWS]
```

I  think it makes sense to adhere to the RFC and at least include a left and right part and an at sign, which however can be done indepent from this PR - so the main question remains, if you consider an integer enough given the fact that there is already rather unique information in the message id with the action id and the per-execution Wid.

Reply:
> so no need for more data from my point of view

fair enough .. though I am not sure if I agree with the assertion that random UUIDs are too slow for this workflow. 

To avoid sending negative numbers you may want to consider masking the result with bitwise operator like `incrementAndGet() & 0xffffff` similar to [TimeBasedUUIDGenerator](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/common/TimeBasedUUIDGenerator.java#L61). It doesn't prevent the internal counter from going into the negative but does prevent the output from going negative and makes it clear that going negative is OK and is part of the design.

Reply:
I found another solution. Using `AtomicLong` in combination with `Long.toUnsignedString()` - that seems easier to read to me than using the bitweise comparison.